Building `cuda.bindings` in parallel might fail non-deterministically

Our new CI (#267) immediately exposes a longstanding, widely known `setuptools`/`distutils` issue when `PARALLEL_LEVEL` is set, xref:
- https://github.com/NVIDIA/cuda-python/actions/runs/12192916020/job/34014323275#step:4:2187
```
    D:\a\cuda-python\cuda-python\cuda_bindings\cuda\bindings\_bindings\loader.cpp : fatal error C1083: Cannot open compiler generated file: 'D:\a\cuda-python\cuda-python\cuda_bindings\build\temp.win-amd64-cpython-310\Release\cuda\bindings\_bindings\loader.obj': Permission denied
```
- https://github.com/pypa/setuptools/issues/3119
- https://github.com/pypa/setuptools/issues/3942
- https://github.com/pypa/setuptools/issues/2442
- scipy/scipy#7112

--- a/cuda_bindings/setup.py
+++ b/cuda_bindings/setup.py
@@ -264,10 +264,12 @@ def do_cythonize(extensions):
 
 sources_list = [
     # private
-    ["cuda/bindings/_bindings/*.pyx", "cuda/bindings/_bindings/loader.cpp"],
+    ["cuda/bindings/_bindings/cydriver.pyx", "cuda/bindings/_bindings/loader.cpp"],
+    ["cuda/bindings/_bindings/cynvrtc.pyx"],
     # utils
-    ["cuda/bindings/_lib/*.pyx", "cuda/bindings/_lib/param_packer.cpp"],
-    ["cuda/bindings/_lib/cyruntime/*.pyx"],
+    ["cuda/bindings/_lib/utils.pyx", "cuda/bindings/_lib/param_packer.cpp"],
+    ["cuda/bindings/_lib/cyruntime/cyruntime.pyx"],
+    ["cuda/bindings/_lib/cyruntime/utils.pyx"],
     # public
     ["cuda/bindings/*.pyx"],
     # public (deprecated, to be removed)

